cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/DelayedHttpServiceResponder.java

```diff
@@ -176,8 +177,11 @@ private void send(int status, ByteBuffer content, String contentType,
    * Since calling one of the send methods multiple times logs a warning, upon transaction failures this
    * method is called to allow setting the failure response without an additional warning.
    */
-  public void setTransactionFailureResponse() {
-    ByteBuffer buffer = Charsets.UTF_8.encode("Transaction failure when committing changes. Aborted transaction.");
+  public void setTransactionFailureResponse(Throwable t) {
+    Throwable rootCause = Throwables.getRootCause(t);
+    ByteBuffer buffer = Charsets.UTF_8.encode("Exception occurred while handling request: "
```

Comment:
should we say "Aborted transaction due to exception ..." In that way we also pass the information that the transaction is aborted.